src/placement.c

```diff
@@ -0,0 +1,513 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include "common/macros.h"
+#include "labwc.h"
+#include "placement.h"
+#include "ssd.h"
+#include "view.h"
+
+#define overlap_bitmap_index(bmp, i, j) \
+	(bmp)->grid[i * ((bmp)->ncols - 1) + j]
+
+static int
+compare_ints(const void *a, const void *b)
+{
+	return *(const int *)a - *(const int *)b;
+}
+
+static void placement_destroy_bitmap(struct overlap_bitmap *bmp)
+{
+	assert(bmp);
+
+	if (bmp->rows) {
+		free(bmp->rows);
+	}
+
+	bmp->nrows = 0;
+	bmp->ncols = 0;
+	bmp->rows = NULL;
+	bmp->cols = NULL;
+	bmp->grid = NULL;
+}
+
+/* Count the number of views on view->output, excluding *view itself */
+static int
+placement_count_views(struct view *view)
```

Comment:
We could drop the `placement_` prefix for static functions (within the translation unit) if you want. I'm easy.